modules/consensus/accept_reorg_test.go

```diff
@@ -0,0 +1,377 @@
+// and cstMain is reorg'd back to have all of the original blocks.
+func (rs *reorgSets) fullReorg() {
+	rs.save()
+	rs.extend()
+	rs.restore()
+}
+
+// TestIntegrationSimpleReorg tries to reorganize a simple block out of, and
+// then back into, the consensus set.
+func TestIntegrationSimpleReorg(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	rs := createReorgSets("TestIntegrationSimpleReorg")
+
+	// Give a simple block to cstMain.
+	rs.cstMain.testSimpleBlock()
+
+	// Try to trigger consensus inconsistencies by doing a full reorg on the
+	// simple block.
+	rs.fullReorg()
+}
+
+// TestIntegrationSiacoinReorg tries to reorganize a siacoin output block out
+// of, and then back into, the consensus set.
+func TestIntegrationSiacoinReorg(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	rs := createReorgSets("TestIntegrationSiacoinReorg")
+
+	// Give a siacoin block to cstMain.
+	rs.cstMain.testSpendSiacoinsBlock()
+
+	// Try to trigger consensus inconsistencies by doing a full reorg on the
+	// simple block.
+	rs.fullReorg()
+}
+
+// TestIntegrationValidStorageProofReorg tries to reorganize a valid storage
+// proof block out of, and then back into, the consensus set.
+func TestIntegrationValidStorageProofReorg(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	rs := createReorgSets("TestIntegrationValidStorageProofReorg")
+
+	// Give a series of blocks containing a file contract and a valid storage
+	// proof to cstMain.
+	rs.cstMain.testValidStorageProofBlocks()
+
+	// Try to trigger consensus inconsistencies by doing a full reorg on the
+	// simple block.
+	rs.fullReorg()
+}
+
+// TestIntegrationMissedStorageProofReorg tries to reorganize a valid storage
+// proof block out of, and then back into, the consensus set.
+func TestIntegrationMissedStorageProofReorg(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	rs := createReorgSets("TestIntegrationMissedStorageProofReorg")
+
+	// Give a series of blocks containing a file contract and a valid storage
+	// proof to cstMain.
+	rs.cstMain.testMissedStorageProofBlocks()
+
+	// Try to trigger consensus inconsistencies by doing a full reorg on the
+	// simple block.
+	rs.fullReorg()
+}
+
+// TestIntegrationComplexReorg stacks up blocks of all types into a single
+// blockchain that undergoes a massive reorg as a stress test to the codebase.
+func TestIntegrationComplexReorg(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	rs := createReorgSets("TestIntegrationComplexReorg")
+
+	// Give a wide variety of block types to cstMain.
+	for i := 0; i < 3; i++ {
+		rs.cstMain.testBlockSuite()
+	}
+	// Give fewer blocks to cstAlt, while still using the same variety.
+	for i := 0; i < 2; i++ {
+		rs.cstAlt.testBlockSuite()
+	}
+
+	// Try to trigger consensus inconsistencies by doing a full reorg on the
+	// simple block.
+	rs.fullReorg()
+}
+
+/// BREAK ///
+/// BREAK ///
+/// BREAK ///
```

Comment:
I could probably document that better. It's a break between the deprecated functions and the not-deprecated functions